packages/rs-drive-abci/src/abci/config.rs

```diff
@@ -0,0 +1,167 @@
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+
+use super::messages::{RequiredIdentityPublicKeysSet, SystemIdentityPublicKeys};
+
+/// AbciAppConfig stores configuration of the ABCI Application.
+#[allow(dead_code)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct AbciConfig {
+    /// Address to listen on
+    ///
+    /// Address should be an URL with scheme `tcp://` or `unix://`, for example:
+    /// - `tcp://127.0.0.1:1234`
+    /// - `unix:///var/run/abci.sock`
+    #[serde(rename = "abci_bind_address")]
+    pub bind_address: String,
+
+    /// Public keys used for system identity
+    #[serde(flatten)]
+    pub keys: Keys,
+
+    /// Height of genesis block; defaults to 1
+    #[serde(default = "AbciConfig::default_genesis_height")]
+    pub genesis_height: u64,
+
+    /// Height of core at genesis
+    #[serde(default = "AbciConfig::default_genesis_core_height")]
+    pub genesis_core_height: u32,
+
+    /// Chain ID of the network to use
+    #[serde(default)]
+    pub chain_id: String,
+}
+
+impl AbciConfig {
+    pub(crate) fn default_genesis_height() -> u64 {
+        1
+    }
+
+    pub(crate) fn default_genesis_core_height() -> u32 {
+        1
+    }
+}
+
+impl FromEnv for AbciConfig {}
+
+#[serde_as]
+#[derive(Clone, Debug, Serialize, Deserialize)]
+
+/// Struct to easily load from environment keys used by the Platform.
+///
+/// Once loaded, Keys can be easily converted to [SystemIdentityPublicKeys]
+///
+/// [SystemIdentityPublicKeys]: super::messages::SystemIdentityPublicKeys
+pub struct Keys {
+    // dpns contract
+    /// hex-encoded
+    #[serde_as(as = "serde_with::hex::Hex")]
+    dpns_master_public_key: Vec<u8>,
+    /// hex-encoded
+    #[serde_as(as = "serde_with::hex::Hex")]
+    dpns_second_public_key: Vec<u8>,
+
+    // dashpay contract
+    /// hex-encoded
+    #[serde_as(as = "serde_with::hex::Hex")]
+    dashpay_master_public_key: Vec<u8>,
+    /// hex-encoded
+    #[serde_as(as = "serde_with::hex::Hex")]
+    dashpay_second_public_key: Vec<u8>,
+
+    // feature flags contract
+    /// hex-encoded
+    #[serde_as(as = "serde_with::hex::Hex")]
+    feature_flags_master_public_key: Vec<u8>,
+    /// hex-encoded
+    #[serde_as(as = "serde_with::hex::Hex")]
+    feature_flags_second_public_key: Vec<u8>,
+
+    // masternode reward shares contract
+    /// hex-encoded
+    #[serde_as(as = "serde_with::hex::Hex")]
+    masternode_reward_shares_master_public_key: Vec<u8>,
+    /// hex-encoded
+    #[serde_as(as = "serde_with::hex::Hex")]
+    masternode_reward_shares_second_public_key: Vec<u8>,
+
+    // withdrawals contract
+    /// hex-encoded
+    #[serde_as(as = "serde_with::hex::Hex")]
+    withdrawals_master_public_key: Vec<u8>,
+    /// hex-encoded
+    #[serde_as(as = "serde_with::hex::Hex")]
+    withdrawals_second_public_key: Vec<u8>,
+}
+
+impl Keys {
+    /// Create new random keys for a given seed
+    pub fn new_random_keys_with_seed(seed: u64) -> Self {
```

Comment:
I think it's worth specifying that this is for testing purposes. I don't think anybody is going to use this in any other scenario, but just to be fool-proof